server/migrator/Importer.java

```diff
@@ -0,0 +1,311 @@
+            attributeCount++;
+            mayCommit();
+        } else {
+            throw GraknException.of(TYPE_NOT_FOUND, relabel(attributeMsg.getLabel()), attributeMsg.getLabel());
+        }
+    }
+
+    private void insertOwnedAttributesThatExist(Thing thing, List<DataProto.Item.OwnedAttribute> ownedMsgs) {
+        List<String> missingOwnerships = new ArrayList<>();
+        for (DataProto.Item.OwnedAttribute ownedMsg : ownedMsgs) {
+            Thing attrThing = getThing(ownedMsg.getId());
+            if (attrThing != null) {
+                thing.setHas(attrThing.asAttribute());
+                ownershipCount++;
+            } else {
+                missingOwnerships.add(ownedMsg.getId());
+            }
+            mayCommit();
+        }
+        this.missingOwnerships.add(new Pair<>(thing.getIID(), missingOwnerships));
+    }
+
+    private void insertMissingOwnerships() {
+        for (Pair<byte[], List<String>> ownership : missingOwnerships) {
+            Thing thing = tx.concepts().getThing(ownership.first());
+            for (String originalAttributeId : ownership.second()) {
+                Thing attrThing = getThing(originalAttributeId);
+                assert thing != null && attrThing != null;
+                thing.setHas(attrThing.asAttribute());
+                ownershipCount++;
+            }
+            mayCommit();
+        }
+        missingOwnerships.clear();
+    }
+
+    private void insertMissingRolePlayers() {
+        for (Pair<byte[], List<Pair<String, List<String>>>> rolePlayers : missingRolePlayers) {
+            Thing thing = tx.concepts().getThing(rolePlayers.first());
+            assert thing != null;
+            Relation relation = thing.asRelation();
+            for (Pair<String, List<String>> pair : rolePlayers.second()) {
+                Map<String, RoleType> roles = getScopedRoleTypes(relation.getType());
+                RoleType role = roles.get(pair.first());
+                assert role != null;
+                for (String originalPlayerId : pair.second()) {
+                    Thing player = getThing(originalPlayerId);
+                    relation.addPlayer(role, player);
+                    playerCount++;
+                }
+            }
+            mayCommit();
+        }
+        missingRolePlayers.clear();
+    }
+
+    private Thing getThing(String originalId) {
+        byte[] newId = idMap.get(originalId);
+        return newId != null ? tx.concepts().getThing(newId) : null;
+    }
+
+    private Map<String, RoleType> getScopedRoleTypes(RelationType relationType) {
+        return relationType.getRelates().collect(
+                Collectors.toMap(x -> x.getLabel().scopedName(), x -> x));
+    }
+
+    private String relabel(String label) {
+        return remapLabels.getOrDefault(label, label);
+    }
+
+    private void mayCommit() {
+        commitWriteCount++;
+        if (commitWriteCount >= BATCH_SIZE) {
```

Comment:
i think `txWriteCount` is clearer name as it signals that its used only for lifetime of one commit?